### ISSUE TITLE
Tor error message at shutdown

### ISSUE THREAD

**Issue by ManfredKarrer** [bisq-network/bisq#2465]
I get often that error log at app shutdown. @freimair could you have a look into it?

```
 12:35:23.334 [NetworkNode:SendMessage-to-yf4mvtsiwzq5hhzj.onion:9999] INFO  b.n.p2p.network.NetworkNode: onFailure at sendMessage: peersNodeAddress=yf4mvtsiwzq5hhzj.onion:9999
	message=GetPeersRequest
	throwable=java.io.IOException: Cannot connect to hidden service 
net.freehaven.tor.control.TorControlError: Error reply: Unknown Onion Service id
	at net.freehaven.tor.control.TorControlConnection.sendAndWaitForResponse(TorControlConnection.java:195)
	at net.freehaven.tor.control.TorControlConnection.destroyHiddenService(TorControlConnection.java:966)
	at org.berndpruenster.netlayer.tor.Tor.unpublishHiddenService(Tor.kt:299)
	at org.berndpruenster.netlayer.tor.NativeTor.shutdown(NativeTor.kt:128)
	at bisq.network.p2p.network.TorNetworkNode.lambda$torNetworkNodeShutDown$3(TorNetworkNode.java:192)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call$$$capture(Executors.java:514)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:111)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:58)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:75)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:844)
```

**Comment by ripcurlx** [comment]
@freimair I also see this every now and then. Is this a problem or an expected behavior? If it is expected we probably shouldn't print this exception.

**Comment by freimair** [comment]
I strongly suspect tor getting cut off by the OS before the Bisq app is ready to do so - hence, the Bisq app tries to shutdown a hidden service that has already been shut down.

I recently tried to make some changes to the netlayer stack to better handle this kind of error. Made some progress, included it with the last tor update which promptly got reverted because windows clients always cause issues - and I have not been able to reproduce them - so I have not been able to "fix" that.

The thing is, the netlayer stack (including interfacing) needs love. Things are not exactly cleanly handled there. Lots of different timeouts, different handling for Windows OS, spagetti code everywhere, nasty stuff - it works somehow, but it is a minefield - touch something and you burn yourself.

I suggest creating some room in budgeting so the netlayer stack can get the love it needs. I will sketch something up...
